core/src/main/java/c/org/rajawali3d/surface/gles/BaseGlesSurfaceRenderer.java

```diff
@@ -1,30 +1,35 @@
 package c.org.rajawali3d.surface.gles;
 
+import c.org.rajawali3d.control.CoreControl;
+import c.org.rajawali3d.control.RenderContextType;
+import c.org.rajawali3d.control.RenderControlClient;
+import c.org.rajawali3d.control.RenderSurfaceView;
+import c.org.rajawali3d.control.gles.GlesRenderControl;
+import c.org.rajawali3d.gl.Capabilities;
+import c.org.rajawali3d.surface.SurfaceRenderer;
+
 import android.content.Context;
 import android.opengl.GLES20;
-import c.org.rajawali3d.core.ARenderControl;
-import c.org.rajawali3d.core.RenderContextType;
-import c.org.rajawali3d.core.RenderControlClient;
-import c.org.rajawali3d.core.RenderSurfaceView;
-import c.org.rajawali3d.gl.Capabilities;
+
 import org.rajawali3d.util.RajLog;
 
 import java.util.Locale;
 
 /**
- * Shared implementation for GL ES extensions of {@link ARenderControl}
+ * Shared implementation for GLES {@link SurfaceRenderer} extensions of {@link CoreControl}
  *
  * @author Randy Picolet
  */
-abstract class GLESRenderer extends ARenderControl {
 
-    GLESRenderer(Context context, RenderSurfaceView renderSurfaceView,
-                 RenderControlClient renderControlClient, double initialFrameRate) {
+abstract class BaseGlesSurfaceRenderer extends CoreControl implements GlesRenderControl {
```

Comment:
`CoreControl` implements `RenderControl` while `GlesRenderControl` extends it. We need to avoid this cycle.

Reply:
Agreed, its wrong will fix.